### PR TITLE
Update node label prefix

### PR DESCRIPTION
**Pull request by kartikjoshi21**
Update node label from node-role.kubernetes.io to node.kubernetes.io

Fixes: #780

### CODE REVIEW

**Review by bpradipt — COMMENTED** [review]
Overall this lgtm. It's a simple change.
However one thing which I'm unsure is whether we need to make changes to the CoCo operator as well.
@wainersm @stevenhorsman ?

**Comment by kartikjoshi21** [comment]
> Overall this lgtm. It's a simple change. However one thing which I'm unsure is whether we need to make changes to the CoCo operator as well. @wainersm @stevenhorsman ?

Yess looks like it need to be changed in operator repo as well. Some places are not related to this but its always good to be on latest KUBE version norms. I will submit a PR in that repo as well

**Comment by kartikjoshi21** [comment]
> Yess looks like it need to be changed in operator repo as well. Some places are not related to this but its always good to be on latest KUBE version norms. I will submit a PR in that repo as well

https://github.com/confidential-containers/operator/pull/195 added this PR In operator repo

**Comment by kartikjoshi21** [comment]
@surajssd  @bpradipt Can we plan to review and merge this and corresponding operator one tagged in this PR ?

File: test/provisioner/provision_azure.go [hunk 1]
```diff
@@ -297,8 +297,8 @@ func (p *AzureCloudProvisioner) CreateCluster(ctx context.Context, cfg *envconf.
 
 	cfg.WithKubeconfigFile(kubeconfigPath)
 
-	// Use cli to label nodes until label is changed from "node-role.kubernetes.io/worker" to "node.kubernetes.io/worker"
-	cmd := exec.Command("kubectl", "label", "nodes", "--all", fmt.Sprintf("%s=%s", "node-role.kubernetes.io/worker", ""))
+	// Update this to use label wile provisioning cluster
```
Comment (surajssd) [inline test/provisioner/provision_azure.go:300]:
```suggestion
	// Update this to use label while provisioning cluster
```

nit: 😉

**Comment by surajssd** [comment]
The tests on the operator are failing for some reason. IMO both of these should be merged at the same time.

**Comment by kartikjoshi21** [comment]
> The tests on the operator are failing for some reason. IMO both of these should be merged at the same time.

Yes, iam not able to open the failing reason in that ci. Looks like these two needs to be merged together.

**Comment by kartikjoshi21** [comment]
> @kartikjoshi21 mind to wait this change get merged into confidential-containers/operator first?

Added hold label until we get operator one merged. Thanks

**Comment by stevenhorsman** [comment]
@kartikjoshi21 - The CoCo version has been merge merged now, and this is quite old, so can you rebase and we can try and get this side in soon? Thanks

File: install/yamls/ccruntime-peer-pods.yaml
Comment (stevenhorsman) [inline install/yamls/ccruntime-peer-pods.yaml:1]: This file shouldn't have been re-added

**Review by stevenhorsman — CHANGES_REQUESTED** [review]

File: install/yamls/ccruntime-peer-pods.yaml
Comment (kartikjoshi21) [reply install/yamls/ccruntime-peer-pods.yaml:1]: Updated in last commit

**Review by stevenhorsman — APPROVED** [review]
LGTM and I've tried it out locally and it resolves the problems we have with mismatch between CAA code and the upstream coco operator. Thanks

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @kartikjoshi21

**Comment by wainersm** [comment]
@stevenhorsman what version of the cloud-api-adaptor image have you used to test this change?

I just ran the libvirt e2e tests on my machine and the latest image (`quay.io/confidential-containers/cloud-api-adaptor:latest`) is still requiring the old label:

```
[wmoschet@wainer-laptop cloud-api-adaptor]$ kubectl get ds -n confidential-containers-system
NAME                             DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                            AGE
cc-operator-daemon-install       1         1         1       1            1           node.kubernetes.io/worker=               12m
cc-operator-daemon-uninstall     0         0         0       0            0           katacontainers.io/kata-runtime=cleanup   13m
cc-operator-pre-install-daemon   1         1         1       1            1           node.kubernetes.io/worker=               13m
cloud-api-adaptor-daemonset      0         0         0       0            0           node-role.kubernetes.io/worker=          13m
```

**Comment by wainersm** [comment]
> @stevenhorsman what version of the cloud-api-adaptor image have you used to test this change?
> 
> I just ran the libvirt e2e tests on my machine and the latest image (`quay.io/confidential-containers/cloud-api-adaptor:latest`) is still requiring the old label:
> 
> ```
> [wmoschet@wainer-laptop cloud-api-adaptor]$ kubectl get ds -n confidential-containers-system
> NAME                             DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                            AGE
> cc-operator-daemon-install       1         1         1       1            1           node.kubernetes.io/worker=               12m
> cc-operator-daemon-uninstall     0         0         0       0            0           katacontainers.io/kata-runtime=cleanup   13m
> cc-operator-pre-install-daemon   1         1         1       1            1           node.kubernetes.io/worker=               13m
> cloud-api-adaptor-daemonset      0         0         0       0            0           node-role.kubernetes.io/worker=          13m
> ```

Never mind, the branch I was working one is outdated.

**Pull request merged**